test/functional/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyOSRTest.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ */
+package com.ibm.j9.jsr292.indyn;
+
+import org.openj9.test.util.VersionCheck;
+
+import org.testng.annotations.Test;
+import org.testng.annotations.DataProvider;
+
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+import org.objectweb.asm.*;
+import static org.objectweb.asm.Opcodes.*;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.ConstantCallSite;
+import java.lang.invoke.CallSite;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+
+public class IndyOSRTest {
+
+	private static Throwable thrower = null;
+	private static String dummyClassPathName = "com/ibm/j9/jsr292/indyn/TestBSMError";
+	private static String dummyClassFullName = "com.ibm.j9.jsr292.indyn.TestBSMError";
+
+	// generate method with invokedynamic bytecode that uses the BSM "bootstrap" below
+	private static byte[] generate(int version) {
+		ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS);
+		MethodVisitor mv;
+		cw.visit(VersionCheck.major() + V1_8 - 8, ACC_PUBLIC, dummyClassPathName + version, null, "java/lang/Object", null);
+
+		mv = cw.visitMethod(ACC_PUBLIC | ACC_STATIC, "dummy", "()V", null, null);
+		mv.visitCode();
+
+		Handle bsm = new Handle(
+			H_INVOKESTATIC,
+			"com/ibm/j9/jsr292/indyn/IndyOSRTest",
+			"bootstrap",
+			"(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;)Ljava/lang/invoke/CallSite;",
+			false
+		);
+
+		mv.visitFieldInsn(GETSTATIC, "java/lang/System", "out", "Ljava/io/PrintStream;");
+		mv.visitLdcInsn(1L);
+		mv.visitLdcInsn(2L);
+		//mv.visitVarInsn(ILOAD, 0);
+		mv.visitLdcInsn(3);
+		mv.visitLdcInsn(4);
+		mv.visitInvokeDynamicInsn("someName", "(JJII)Ljava/lang/String;", bsm);
+		mv.visitMethodInsn(INVOKEVIRTUAL, "java/io/PrintStream", "println", "(Ljava/lang/String;)V", false);
+		mv.visitInsn(RETURN);
+		mv.visitMaxs(6, 4);
+		mv.visitEnd();
+		cw.visitEnd();
+		return cw.toByteArray();
+	}
+
+
+
+	public static CallSite bootstrap(MethodHandles.Lookup lookup, String name, MethodType mt) throws Throwable {
+		if (thrower == null) {
+			MethodHandle target = MethodHandles.lookup().findStatic(IndyTest.class, "sanity", MethodType.methodType(String.class, long.class, long.class, int.class, int.class));
```

Comment:
I think you mean to refer to this class (`IndyOSRTest`).
Please fold this long line:
```java
		MethodHandle target = MethodHandles.lookup().findStatic(
				IndyOSRTest.class,
				"sanity",
				MethodType.methodType(String.class, long.class, long.class, int.class, int.class));
```